javascript/src/components/streams/StreamControls.jsx

```diff
@@ -0,0 +1,86 @@
+/* global jsRoutes */
+
+'use strict';
+
+var React = require('react/addons');
+var DropdownButton = require('react-bootstrap').DropdownButton;
+var MenuItem = require('react-bootstrap').MenuItem;
+var ButtonGroup = require('react-bootstrap').ButtonGroup;
+var StreamForm = require('./StreamForm');
+var PermissionsMixin = require('../../util/PermissionsMixin');
+var UsersStore = require('../../stores/users/UsersStore');
+
+var StreamControls = React.createClass({
+    mixins: [PermissionsMixin],
+    getInitialState() {
+        return {};
+    },
+    _onResume(evt) {
+        this.props.onResume(this.props.stream);
+        this.refs.dropdownButton.setDropdownState(false);
+    },
+    _onPause(evt) {
+        this.props.onPause(this.props.stream);
+        this.refs.dropdownButton.setDropdownState(false);
+    },
+    _onEdit(evt) {
+        this.refs.streamForm.open();
+    },
+    _onClone(evt) {
+        this.refs.cloneForm.open();
+    },
+    _onCloneSubmit(streamId, stream) {
+        this.props.onClone(this.props.stream.id, stream);
+    },
+    _onQuickAdd() {
+        this.props.onQuickAdd(this.props.stream.id);
+    },
+    componentDidMount() {
+        UsersStore.load(this.props.username).done((user) => {
```

Comment:
If by doing it globally you meant to render the user into the play template, and then use it in the component, I think it would be a good solution for now. I would leave any refactoring of other components out, as it would be out of the scope of this PR, and may also require more time that we should probably expend on it at the moment.